strr-api/src/strr_api/resources/application.py

```diff
@@ -125,6 +126,8 @@ def create_application(application_number: Optional[str] = None):
                     message=ErrorMessage.APPLICATION_NOT_MODIFIABLE.value,
                     http_status=HTTPStatus.BAD_REQUEST,
                 )
+            # update application date
+            application.application_date = datetime.now(timezone.utc)
```

Comment:
alternatively we could have this be inside `if not is_draft:` so that it only updates when the application is completed.

You may also prefer this is moved to the application service, but I thought this might be simpler / its easy to know it only affects drafts